semant.ml

```diff
@@ -55,9 +55,12 @@ let check (program) =
       let same = t1 = t2 in
       (* Determine expression type based on operator and operand types *)
       let ty = match op with
-        Add | Sub | Mul | Div when same && t1 = Int   -> Int
-      | Add | Sub | Mul | Div when same && t1 = Float   -> Float
+        Add | Sub | Mul | Div | Mod when same && t1 = Int   -> Int
+      | Add | Sub | Mul | Div | Mod when same && t1 = Float   -> Float
       | Add | Sub when same && t1 = Matrix -> Matrix
+      | Eq | Neq  when same -> Bool
+      | Less | Leq | Greater | Geq when same && (t1 = Int || t1 = Float) -> Bool

```

Comment:
Are we planning on allowing comparisons between int and float?

Reply:
As stated in the comment, "All binary operators require operands of the same type". So the short answer is no.